src/ComparedQueryable/NativeQueryable/EnumerableRewriter.cs

```diff
@@ -54,13 +56,13 @@ protected override Expression VisitMethodCall(MethodCallExpression m)
             return m;
         }
 
-        private ReadOnlyCollection<Expression> FixupQuotedArgs(MethodInfo mi, ReadOnlyCollection<Expression> argList)
+        protected virtual ReadOnlyCollection<Expression> FixupQuotedArgs(MethodInfo mi, ReadOnlyCollection<Expression> argList)
         {
             ParameterInfo[] pis = mi.GetParameters();
             if (pis.Length > 0)
             {
                 List<Expression> newArgs = null;
-                for (int i = 0, n = pis.Length; i < n; i++)
+                for (int i = 0, n = argList.Count; i < n; i++)

```

Comment:
I changed this back. There was a version where I had been playing around with this method, but it's not needed once the method is made virtual.